src/transactions/saga.ts

```diff
@@ -222,7 +227,65 @@ function* watchAddressToE164PhoneNumberUpdate() {
   )
 }
 
+function* watchPendingTransaction(transaction: StandbyTransaction) {
+  const { transactionHash, networkId } = transaction
+
+  if (!transactionHash) {
+    Logger.warn(TAG, `Transaction ${transaction.context.id} has no hash, skipping`)
+    return
+  }
+
+  // TODO: Remove this constraint when our viem client supports other networks
+  const supportedNetworkdIds = [NetworkId['celo-mainnet'], NetworkId['celo-alfajores']]
+  if (!supportedNetworkdIds.includes(networkId)) {
+    Logger.warn(TAG, `Transaction ${transaction.context.id} is not on the celo network, skipping`)
+    return
+  }
+
+  try {
+    const receipt = yield* call(publicClient.celo.getTransactionReceipt, {
+      hash: transactionHash as Hash,
+    })
+
+    if (receipt) {
+      if (receipt.status == 'success') {
+        yield* put(
+          transactionConfirmed(transaction.context.id, {
+            transactionHash: receipt.transactionHash,
+            block: receipt.blockNumber.toString(),
+            status: true,
+          })
+        )
+      }
+
+      if (receipt.status == 'reverted') {
+        yield* put(transactionFailed(transaction.context.id))
+      }
+    }
+  } catch (e) {
+    Logger.error(
+      TAG,
+      `Found error when trying to fetch status for transaction with hash: ${transactionHash}`,
+      e
+    )
+    return
+  }
+}
+
+export function* watchPendingTransactions() {
+  while (true) {
+    const pendingStandbyTransactions: StandbyTransaction[] = yield* select(
+      watchablePendingTransactionSelector
```

Comment:
i'm wondering if we can just use the normal pending tx's selector, since we check for the presence of the tx hash in the watchPendingTransaction saga anyway?